packages/instrumentation-chromadb/src/instrumentation.ts

```diff
@@ -0,0 +1,202 @@
+    this._wrap(module.Collection.prototype, "delete", this.wrapperMethod());
+    this._wrap(module.Collection.prototype, "modify", this.wrapperMethod());
+
+    return module;
+  }
+
+  private unwrap(module: typeof chromadb) {
+    this._unwrap(module.Collection.prototype, "add");
+    this._unwrap(module.Collection.prototype, "get");
+    this._unwrap(module.Collection.prototype, "query");
+    this._unwrap(module.Collection.prototype, "update");
+    this._unwrap(module.Collection.prototype, "upsert");
+    this._unwrap(module.Collection.prototype, "peek");
+    this._unwrap(module.Collection.prototype, "delete");
+    this._unwrap(module.Collection.prototype, "modify");
+
+    return module;
+  }
+
+  private wrapperMethod() {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const plugin = this;
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    return (original: Function) => {
+      return function method(this: any, ...args: any) {
+        const span = plugin._startSpan({
+          params: args[0],
+          methodName: original.name,
+        });
+        const execContext = trace.setSpan(context.active(), span);
+        const execPromise = safeExecuteInTheMiddle(
+          () => {
+            return context.with(execContext, () => {
+              return original.apply(this, args);
+            });
+          },
+          () => null,
+        );
+        const wrappedPromise = plugin._wrapPromise(
+          original.name,
+          span,
+          execPromise,
+        );
+        return context.bind(execContext, wrappedPromise as any);
+      };
+    };
+  }
+
+  private _wrapPromise<T>(
+    methodName: string,
+    span: Span,
+    promise: Promise<T>,
+  ): Promise<T> {
+    return promise
+      .then(async (result) => {
+        const awaitedResult = (await result) as
+          | chromadb.GetResponse
+          | chromadb.QueryResponse;
+
+        this._endSpan({
+          methodName,
+          span,
+          result: awaitedResult,
+        });
+
+        return new Promise<T>((resolve) => {
+          resolve(result);
+        });
+      })
+      .catch((error: Error) => {
+        return new Promise<T>((_, reject) => {
+          span.setStatus({
+            code: SpanStatusCode.ERROR,
+            message: error.message,
+          });
+          span.recordException(error);
+          reject(error);
+        });
+      })
+      .finally(() => {
+        span.end();
+      });
+  }
+
+  private _startSpan({
+    params,
+    methodName,
+  }: {
+    params: chromadb.GetParams | chromadb.PeekParams;
+    methodName: string;
+  }): Span {
+    const attributes: Attributes = {
+      [SpanAttributes.VECTOR_DB_VENDOR]: "ChromaDB",
+    };
+    const span = this.tracer.startSpan(`chromadb.${methodName}`, {
+      kind: SpanKind.CLIENT,
+      attributes,
+    });
+
+    // Instrumenting only for query and peak
+    if (methodName === "query" || methodName === "peek") {
```

Comment:
Events should match [the format we have in Python](https://github.com/traceloop/openllmetry/blob/main/packages/opentelemetry-instrumentation-pinecone/opentelemetry/instrumentation/pinecone/__init__.py) - each vector in a separate event, etc.

Reply:
@nirga I've added event based attributes similar to Python. Please re-review it.

<img width="1181" alt="image" src="https://github.com/traceloop/openllmetry-js/assets/57614947/16aabd97-a089-4e5a-b5c1-594f51b41819">